### PR TITLE
Add TorchLog1pVisitor

### PR DESCRIPTION
**Pull request by kit1980**
Suggest using `torch.log1p(x)` instead of `torch.log(1 + x)`.
Only a checker for now, probably not worth spending time on a codemod as it's not a very common issue.

I used this to create a PR to botorch https://github.com/pytorch/botorch/pull/2539

### CODE REVIEW

File: torchfix/torchfix.py [hunk 1]
```diff
@@ -28,15 +29,16 @@
 
 ALL_VISITOR_CLS = [
     TorchDeprecatedSymbolsVisitor,
```
Comment (kit1980) [inline torchfix/torchfix.py:31]: Any ideas how to avoid this ugly duplication of all visitors?

File: tests/fixtures/misc/checker/log1p.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import torch
+a = torch.randn(5)
+b = torch.log(1 + a)
+c = torch.log(a + 1)
+b = torch.log(1.0 + a)
+c = torch.log(a + 1.0)
```
Comment (sbrugman) [inline tests/fixtures/misc/checker/log1p.py:6]: Could we already add Array API examples here? We don't have the capability of automatically detecting them at the moment, but it's good to be explicit about the known false negatives, especially when we can catch them later

```python
import torch

a = torch.randn(5)

print((a + 1).log())
print(a.log1p())
```

File: torchfix/torchfix.py (hunk 1 above)
Comment (sbrugman) [reply torchfix/torchfix.py:31]: Yes, we can automatically pick them up with a registry pattern. Already got something in progress for this.

**Comment by sbrugman** [comment]
Nice catch, LGTM

**Review by sbrugman — APPROVED** [review]

**Comment by sbrugman** [comment]
Let's also add https://github.com/pytorch/botorch/pull/2541

**Comment by kit1980** [comment]
> Let's also add [pytorch/botorch#2541](https://github.com/pytorch/botorch/pull/2541) / [pytorch/botorch#2540](https://github.com/pytorch/botorch/pull/2540) in a follow-up

I'll add `expm1`. Looks like Botorch guys were hinted by my PR about `log1p`.

File: tests/fixtures/misc/checker/log1p.py (hunk 2 above)
Comment (kit1980) [reply tests/fixtures/misc/checker/log1p.py:6]: Added a false negative.

**Pull request merged**